src/lib/y2partitioner/widgets/blk_device_edit_button.rb

```diff
@@ -29,21 +29,52 @@
 
 module Y2Partitioner
   module Widgets
-    # Button for opening the editing workflow (basically mount and format
-    # options) on a block device.
+    # Button for editing a block device
     class BlkDeviceEditButton < DeviceButton
-      # TRANSLATORS: button label to edit a block device
       def label
+        # TRANSLATORS: button label for editing a block device
         _("Edit...")
       end
 
     private
 
       # @see DeviceButton#actions
+      # When the device is a disk, dasd, multipath or bios raid, edit means
+      # to jump to the tree page of that device. For partition or software
+      # raid, the editing workflow to select mount and format options is shown.
       def actions
         UIState.instance.select_row(device.sid)
-        actions_result = Actions::EditBlkDevice.new(device).run
-        result(actions_result)
+        partition? || software_raid? ? super : go_to_disk_page
+      end
+
+      # Whether the device is a partition
+      #
+      # @return [Booelan]
+      def partition?
+        device.is?(:partition)
+      end
+
+      # Whether the device is a software raid
+      #
+      # @return [Booelan]
+      def software_raid?
+        device.is?(:md) && device.software_defined?
```

Comment:
Probably not 100% related, but this makes me think that `Md` and `Dm` objects should return true to `device.is?(:raid)`. Just thinking aloud.

Reply:
Yes, it makes sense.